mantidimaging/core/fitting/fitting_engine.py

```diff
@@ -22,17 +22,44 @@ def get_parameter_names(self) -> list[str]:
     def get_init_params_from_roi(self, region: FittingRegion) -> dict[str, float]:
         return self.model.get_init_params_from_roi(region)
 
-    def find_best_fit(self, xdata: np.ndarray, ydata: np.ndarray, initial_params: list[float]) -> dict[str, float]:
-        additional_params = self.model.prefitting(xdata, ydata, initial_params)
-        params_to_fit = initial_params[:len(initial_params) - len(additional_params)]
+    def find_best_fit(
+        self,
+        xdata: np.ndarray,
+        ydata: np.ndarray,
+        initial_params: list[float],
+        sigma: np.ndarray | None = None,
+    ) -> tuple[dict[str, float], float, float]:
+        """
+        Perform a model fit and return the fitted parameters, unweighted SSE,
+        and sigma-weighted SSE.
 
-        def f(params_to_fit):
-            if additional_params:
-                params_to_fit = np.concatenate((params_to_fit, np.array(additional_params)), axis=None)
-            return ((self.model.evaluate(xdata, params_to_fit) - ydata)**2).sum()
+        :param xdata: TOF data (independent variable)
+        :param ydata: Spectrum data (dependent variable)
+        :param initial_params: Initial parameter estimates
+        :param sigma: Optional per-point uncertainties (same length as ydata).
+                      If None, fall back to sqrt(ydata) with floor at 1.
+        :return: (fit_params, sse, weighted_sse)
+        """
+        additional_params: list[float] = self.model.prefitting(xdata, ydata, initial_params)
+        fit_param_count = len(initial_params) - len(additional_params)
+        params_to_fit: np.ndarray = np.array(initial_params[:fit_param_count])
 
-        result = minimize(f, params_to_fit, method="Nelder-Mead")
+        def chi_squared(params_subset: list[float]) -> float:
+            full_params = list(params_subset) + additional_params
+            residuals = self.model.evaluate(xdata, full_params) - ydata
+            return float(np.sum(residuals**2))
 
+        result = minimize(chi_squared, params_to_fit, method="Nelder-Mead")
         all_param_names = self.model.get_parameter_names()
-        all_params = list(result.x) + additional_params
-        return dict(zip(all_param_names, all_params, strict=True))
+        all_params: list[float] = list(result.x) + additional_params
+        fit_params: dict[str, float] = dict(zip(all_param_names, all_params, strict=True))
+        final_residuals = self.model.evaluate(xdata, all_params) - ydata
```

Comment:
`result.fun` should have the final value of the objective function, which is already the sum of the squares.